### PR TITLE
Add queuejob reference manager

### PR DESCRIPTION
**Pull request by hanghliu**
Add queuejob reference manager to manage the membership between queuejob resources and its members.

### CODE REVIEW

File: pkg/controller/queuejobresources/pod/pod.go [hunk 1]
```diff
@@ -176,11 +179,12 @@ func (qjrPod *QueueJobResPod) scaleUpQueueJobResource(resIndex int, diff int32,
 		return err
 	}
 
-	if template.Template.Labels == nil {
-
-		template.Template.Labels = map[string]string{}
+	//TODO: need set reference after Pod has been really dded
```
Comment (jinzhejz) [inline pkg/controller/queuejobresources/pod/pod.go]: seems a typo `dded`

File: pkg/controller/queuejob_controller.go [hunk 2]
```diff
@@ -279,8 +285,19 @@ func (qjm *QueueJobController) syncQueueJob(key string) error {
 	}
 
 	if job.Spec.AggrResources.Items != nil {
-		for i, ar := range job.Spec.AggrResources.Items {
-			qjm.qjobResControls[ar.Type].Sync(i, sharedJob, &ar)
+		for i := range job.Spec.AggrResources.Items {
+			qjm.refManager.AddTag(&job.Spec.AggrResources.Items[i], func() string {
```
Comment (jinzhejz) [inline pkg/controller/queuejob_controller.go]: suggest adding error handling for `AddTag` when the return value is not nil

File: pkg/controller/queuejobresources/pod/pod.go [hunk 3]
```diff
@@ -176,11 +179,12 @@ func (qjrPod *QueueJobResPod) scaleUpQueueJobResource(resIndex int, diff int32,
...
+	tmpPod := v1.Pod{}
+	qjrPod.refManager.AddReference(qjobRes, &tmpPod)
```
Comment (jinzhejz) [inline pkg/controller/queuejobresources/pod/pod.go]: suggest adding error handling for `AddReference` when the return value is not nil

File: pkg/controller/queuejob_controller.go (hunk 2 above)
Comment (hanghliu) [reply pkg/controller/queuejob_controller.go]: ok

File: pkg/controller/queuejobresources/pod/pod.go (hunk 3 above)
Comment (hanghliu) [reply pkg/controller/queuejobresources/pod/pod.go]: ok

File: pkg/controller/queuejobresources/pod/pod.go (hunk 1 above)
Comment (hanghliu) [reply pkg/controller/queuejobresources/pod/pod.go]: Yes, it should be added

**Review by jinzhejz — APPROVED** [review]
lgtm

**Pull request merged**